src/modules/mc_pos_control/PositionControl/PositionControl.hpp

```diff
@@ -121,66 +120,42 @@ class PositionControl
 	void setHoverThrust(const float thrust) { _hover_thrust = thrust; }
 
 	/**
-	 * Update the current vehicle state.
+	 * Pass the current vehicle state to the controller
 	 * @param PositionControlStates structure
 	 */
-	void updateState(const PositionControlStates &states);
+	void setState(const PositionControlStates &states);
 
 	/**
-	 * Update the desired setpoints.
+	 * Pass the desired setpoints
+	 * Note: NAN value means no feed forward/leave state uncontrolled if there's no higher order setpoint.
 	 * @param setpoint a vehicle_local_position_setpoint_s structure
-	 * @return true if setpoint has updated correctly
+	 * @return true if a valid setpoint was set
 	 */
-	bool updateSetpoint(const vehicle_local_position_setpoint_s &setpoint);
+	bool setInputSetpoint(const vehicle_local_position_setpoint_s &setpoint);
 
 	/**
-	 * Set constraints that are stricter than the global limits.
+	 * Pass constraints that are stricter than the global limits
+	 * Note: NAN value means no constraint, take maximum limit of controller.
 	 * @param constraints a PositionControl structure with supported constraints
 	 */
-	void updateConstraints(const vehicle_constraints_s &constraints);
+	void setConstraints(const vehicle_constraints_s &constraints);
 
 	/**
 	 * Apply P-position and PID-velocity controller that updates the member
 	 * thrust, yaw- and yawspeed-setpoints.
 	 * @see _thr_sp
 	 * @see _yaw_sp
 	 * @see _yawspeed_sp
-	 * @param dt the delta-time
+	 * @param dt time in seconds since last iteration
+	 * @return true if output setpoint is executable, false if not
 	 */
-	void generateThrustYawSetpoint(const float dt);
+	void update(const float dt);
 
 	/**
-	 * 	Set the integral term in xy to 0.
-	 * 	@see _thr_int
+	 * Set the integral term in xy to 0.
+	 * @see _vel_int
 	 */
-	void resetIntegralXY() { _thr_int(0) = _thr_int(1) = 0.f; }
-
-	/**
-	 * 	Set the integral term in z to 0.
-	 * 	@see _thr_int
-	 */
-	void resetIntegralZ() { _thr_int(2) = 0.f; }
-
-	/**
-	 * 	Get the
-	 * 	@see _vel_sp
-	 * 	@return The velocity set-point that was executed in the control-loop. Nan if velocity control-loop was skipped.
-	 */
-	const matrix::Vector3f getVelSp() const
-	{
-		matrix::Vector3f vel_sp{};
-
-		for (int i = 0; i <= 2; i++) {
-			if (_ctrl_vel[i]) {
-				vel_sp(i) = _vel_sp(i);
-
-			} else {
-				vel_sp(i) = NAN;
-			}
-		}
-
-		return vel_sp;
-	}
+	void resetIntegral() { _vel_int = matrix::Vector3f(); }
```

Comment:
```suggestion
	void resetIntegral() { _vel_int.setZero(); }
```

Reply:
Fixed in place: https://github.com/PX4/Firmware/compare/0d2e3463adf676661ebd46ca353df4d969506111..312e10e38d2bf7a891ef24f9afef6c3ccbd7f81a